src/home-page/components/ProfessionCard.tsx

```diff
@@ -41,7 +41,7 @@ export const ProfessionCard: React.FC<CardProps> = ({
   clickHandler,
   route,
 }) => (
-  <CategoryCardContainer onClick={() => clickHandler(route)}>
+  <CategoryCardContainer onClick={() => (clickHandler && route) ? clickHandler(route) : null}>
```

Comment:
This conditional function call is a warning sign that maybe we should take a step back and reconsider if everything is well here. Maybe we could lift the click handler to the parent, that is do not pass it, but keep the logic one level up? Or maybe it's better to duplicate the Component?
Also, could we change the name of the container to `ProfessionCardContainer` (my bad, I missed it when doing the first change)?

Reply:
I refactored ProfessionCard. Please check new changes. :)